packages/styles/scss/components/select/_select.scss

```
@@ -284,27 +284,56 @@
   .#{$prefix}--select--slug .#{$prefix}--slug {
     position: absolute;
     inset-block-start: 50%;
-    inset-inline-end: $spacing-08;
+    inset-inline-end: calc($spacing-08 + 9px);
+    margin-block-start: convert.to-rem(0.5px);
     transform: translateY(-50%);
   }
 
+  .#{$prefix}--select--slug .#{$prefix}--slug::after,
+  .#{$prefix}--select--slug .#{$prefix}--slug::before {
+    position: absolute;
+    background-color: $border-subtle-01;
+    block-size: convert.to-rem(16px);
+    content: '';
+    inline-size: convert.to-rem(1px);
+  }
+
+  .#{$prefix}--select--slug .#{$prefix}--slug::before {
+    display: none;
+    inset-inline-start: convert.to-rem(-9px);
```

```suggestion
    inset-inline-start: calc(-#{$spacing-03} - convert.to-rem(1px));
```

In the spirit of @tay1orjones [PR around magic numbers](https://github.com/carbon-design-system/carbon/pull/15995), would we want to update these values to be the padding value plus the divider size?